x-pack/plugins/security/common/constants.ts

```diff
@@ -29,7 +29,7 @@ export const NEXT_URL_QUERY_STRING_PARAMETER = 'next';
  * - Must contain only letters, numbers, spaces, punctuation and printable symbols.
  * - Must not contain leading or trailing spaces.
  */
-export const NAME_REGEX = /^(?! )[a-zA-Z0-9 !"#$%&'()*+,\-./\\:;<=>?@\[\]^_`{|}~]+(?<! )$/;
+export const NAME_REGEX = /^(?! )[a-zA-Z0-9 !"#$%&'()*+,\-./\\:;<=>?@\[\]^_`{|}~]*[a-zA-Z0-9!"#$%&'()*+,\-./\\:;<=>?@\[\]^_`{|}~]$/;
```

Comment:
__nit:__ hmm, it doesn't look like we have any tests that validate this "no trailing whitespace" requirement or I missed that? If true, would you mind adding some?

Reply:
Cool, I've added that in